packages/extension/src/utils/jsonrpc.ts

```diff
@@ -11,28 +13,37 @@ export const toJsonRpcRequest = (message: RelayMessage): JsonRPCRequest => ({
   params: Array.isArray(message.payload.params) ? message.payload.params : []
 });
 
-export const addChainId = (
+export const addMissingParams = async (
   params: unknown[] | undefined,
-  chainId: number
-): unknown[] | undefined => {
+  state: ApplicationState
+): Promise<unknown[] | undefined> => {
   if (!Array.isArray(params)) {
     return undefined;
   }
 
-  return [{ ...((params[0] as Record<string, unknown>) ?? {}), chainId }];
+  const providers = state.jsonrpc.network.providers;
+  const provider = new FallbackProvider(providers.map((url) => new JsonRpcProvider(url)));
+
+  // @ts-expect-error No type atm
+  const nonce = params[0].nonce ?? hexlify(await provider.getTransactionCount(params[0].from));
```

Comment:
Can you explain?

Reply:
`params` is typed as `unknown` for this function currently

Reply:
Can you use a type guard?